### ISSUE TITLE
Input/Output with Continuous FFT

### ISSUE THREAD

**Issue by joem5636** [spatialaudio/python-sounddevice#387]
wire.py only does input/output and adapting it for FFT filtering is non-trivial. After much research I figured out how to do continuous FFT and have modified wire.py to wireFFT.py which I contribute to "the cause."

[wireFFT.zip](https://github.com/spatialaudio/python-sounddevice/files/7796802/wireFFT.zip)

**Comment by HaHeho** [comment]
Hey @joem5636 thanks for your efforts and the interest on experimenting with the toolbox.

Posting ZIP files for downloading is not good practice. One should not download such from unknown sources. ;)
Directly posting your code would be helpful. Otherwise, you seem to have also have published your code here, right?
https://github.com/joem5636/WireFFT/blob/master/wireFFT.py

**Comment by HaHeho** [comment]
As you have figured out, doing real-time processing in the frequency domain is not trivial in practice. I am also not aware of any popular packages which implement Overlapp-Add-Convolution entirely in Python.

I have implemented Overlapp-Save for example [here](https://github.com/AppliedAcousticsChalmers/ReTiSAR/blob/1033542e56b224f85e8a0fb5ca275561dae061b6/ReTiSAR/_convolver.py#L436) as part of my processing pipeline for spherical microphone arrays. There is a lot of other stuff around it, but you could find all the essential building blocks there. I hope the documentation around it somewhat helps to understand what is going on.

A general comment. The way of data organization i.e., the order in which the time blocks / audio channels / audio samples are stacked in numpy arrays, matters quite a bit in terms of processing performance. Obviously, careful planning of the exact way data is gathered, shifted, multiplied and distributed is crucial for a correct processing without artifacts.

**Comment by joem5636** [comment]
Correct. Posting .py was not allowed and cut-paste lost formatting, hence
the .zip.

**Comment by HaHeho** [comment]
> Correct. Posting .py was not allowed and cut-paste lost formatting, hence the .zip.

True. But you can use Markdown in the comments as well, and therefore use code blocks, see https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#code-and-syntax-highlighting

**Comment by joem5636** [comment]
I suspected I could use <code></code> or the equivalent but knew .zip would
work and i already had the files zipped.

Best,

Joee
